### PR TITLE
Add queue_name for notifications table

### PR DESCRIPTION
**Pull request by jzbahrai**
# Summary | Résumé

We are going to store the name of the queue we send the notification to in the notifications table.

Code was tested by running `flask db upgrade`

```
                            Table "public.notifications"
       Column        |            Type             | Collation | Nullable | Default
---------------------+-----------------------------+-----------+----------+---------
 id                  | uuid                        |           | not null |
 to                  | character varying           |           | not null |
 job_id              | uuid                        |           |          |
 service_id          | uuid                        |           |          |
 template_id         | uuid                        |           |          |
 created_at          | timestamp without time zone |           | not null |
 sent_at             | timestamp without time zone |           |          |
 sent_by             | character varying           |           |          |
 updated_at          | timestamp without time zone |           |          |
 reference           | character varying           |           |          |
 template_version    | integer                     |           | not null |
 job_row_number      | integer                     |           |          |
 _personalisation    | character varying           |           |          |
 api_key_id          | uuid                        |           |          |
 key_type            | character varying(255)      |           | not null |
 notification_type   | notification_type           |           | not null |
 billable_units      | integer                     |           | not null |
 client_reference    | character varying           |           |          |
 international       | boolean                     |           |          |
 phone_prefix        | character varying           |           |          |
 rate_multiplier     | numeric                     |           |          |
 notification_status | text                        |           |          |
 normalised_to       | character varying           |           |          |
 created_by_id       | uuid                        |           |          |
 reply_to_text       | character varying           |           |          |
 postage             | character varying           |           |          |
 provider_response   | text                        |           |          |
 queue_name          | text                        |           |          |
Indexes:
```

Also added the code for notification_history:
```
                        Table "public.notification_history"
       Column        |            Type             | Collation | Nullable | Default
---------------------+-----------------------------+-----------+----------+---------
 id                  | uuid                        |           | not null |
 job_id              | uuid                        |           |          |
 job_row_number      | integer                     |           |          |
 service_id          | uuid                        |           |          |
 template_id         | uuid                        |           |          |
 template_version    | integer                     |           | not null |
 api_key_id          | uuid                        |           |          |
 key_type            | character varying           |           | not null |
 notification_type   | notification_type           |           | not null |
 created_at          | timestamp without time zone |           | not null |
 sent_at             | timestamp without time zone |           |          |
 sent_by             | character varying           |           |          |
 updated_at          | timestamp without time zone |           |          |
 reference           | character varying           |           |          |
 billable_units      | integer                     |           | not null |
 client_reference    | character varying           |           |          |
 international       | boolean                     |           |          |
 phone_prefix        | character varying           |           |          |
 rate_multiplier     | numeric                     |           |          |
 notification_status | text                        |           |          |
 created_by_id       | uuid                        |           |          |
 postage             | character varying           |           |          |
 queue_name          | text                        |           |          |
 ```

### CODE REVIEW

**Review by jimleroyer — COMMENTED** [review]
Can we add the model mapping before and try out if that actually works with unit tests? I worry that once we merge this, this will go in staging and we'll have to deploy this to production, or that will block the release pipeline if we don't want this to propagate.

Also, the notification_history table might need that change as well, including the migration to get there, unless we decide to delay transfer of the new fields for now.

**Comment by jzbahrai** [comment]
@jimleroyer done! can you re-review?

File: migrations/versions/0326_add_queue_notifications.py
Comment (sastels) [inline migrations/versions/0326_add_queue_notifications.py:26]: should the history table drop just be "queue_name"?

**Review by sastels — CHANGES_REQUESTED** [review]
I think the drop command is wrong? Looks good other than that.

**Review by sastels — APPROVED** [review]
tested locally, works for me!

**Comment by jimleroyer** [comment]
Good work! (sorry I am late to the party)

**Comment by josh-cds** [comment]
@jimleroyer @sastels what's the current status of this?

**Comment by jimleroyer** [comment]
@josh-cds This is a pull request and it might appear in ZenHub if you didn't filter to only see issue. I am not sure if we want to treat PRs the same as issues and have them go through all the same columns. It would add some more weight to the process. We can discuss more offline (we could have a parking lot after our daily stand up).

**Pull request merged**
